packages/caliper-fabric/lib/identity-management/IdentityManager.js

```diff
@@ -191,7 +191,14 @@ class IdentityManager {
      * @private
      */
     async _extractIdentitiesFromWallet(mspId, wallet) {
-        // TODO: To be implemented
+        const walletFacade = await this.walletFacadeFactory.create(wallet.path);
+        const allIDNames = await walletFacade.getAllIdentityNames();
+        // eslint-disable-next-line no-console
+        console.log({allIDNames});
+        for (const identityName in allIDNames){
+            const identity = await walletFacade.export(identityName);
+            await this._addToWallet(identity.mspid, allIDNames[identityName], identity.certificate, identity.privateKey);
```

Comment:
with the change to `of` in the for loop, you can replace the `allIDNames[identityName]` with just `identityName`